src/miner_lora_light.erl

```diff
@@ -0,0 +1,799 @@
+    lager:info("Gateway ~p", [lager:pr(Gateway1, ?MODULE)]),
+    Mac = Gateway1#gateway.mac,
+    State1 = maybe_update_gps(Status, State),
+    handle_json_data(maps:remove(<<"stat">>, Map), Gateway1, RxInstantLocal_us,
+                     State1#state{gateways=maps:put(Mac, Gateway1, Gateways)});
+handle_json_data(_, _Gateway, _RxInstantLocal_us, State) ->
+    State.
+
+%% cache GPS the state with each update.  I'm not sure if this will
+%% lead to a lot of wander, but I do want to be able to refine if we
+%% have a poor quality initial lock.  we might want to keep track of
+%% server boot time and lock it down after some period of time.
+-spec maybe_update_gps(#{}, state()) -> state().
+maybe_update_gps(#{<<"lati">> := Lat, <<"long">> := Long}, State) ->
+    State#state{latlong = {Lat, Long}};
+maybe_update_gps(_Status, State) ->
+    State.
+maybe_send_udp_ack(_Socket, _IP, _Port, _Packet, false = _RegDomainConfirmed)->
+    ok;
+maybe_send_udp_ack(Socket, IP, Port, Packet, _RegDomainConfirmed)->
+    ok = gen_udp:send(Socket, IP, Port, Packet).
+
+-spec sort_packets(list()) -> list().
+sort_packets(Packets) ->
+    lists:sort(
+        fun(A, B) ->
+            packet_snr(A) >= packet_snr(B)
+        end,
+        Packets
+    ).
+
+-spec handle_packets(list(), gateway(), integer(), state()) -> state().
+handle_packets([], _Gateway, _RxInstantLocal_us, State) ->
+    State;
+handle_packets(_Packets, _Gateway, _RxInstantLocal_us, #state{reg_domain_confirmed = false} = State) ->
+    State;
+handle_packets([Packet|Tail], Gateway, RxInstantLocal_us, #state{reg_region = _Region} = State) ->
+    POCVersion = application:get_env(miner, poc_version, 11),
+    Data = base64:decode(maps:get(<<"data">>, Packet)),
+    case route(Data) of
+        error ->
+            ok;
+        {onion, Payload} ->
+            Freq = maps:get(<<"freq">>, Packet),
+            %% onion server
+            UseRSSIS = case POCVersion of
+                X when X > 10 -> true;
+                _ -> false
+            end,
+            miner_onion_server_light:decrypt_radio(
+                Payload,
+                erlang:trunc(packet_rssi(Packet, UseRSSIS)),
+                packet_snr(Packet),
+                %% TODO we might want to send GPS time here, if available
+                maps:get(<<"tmst">>, Packet),
+                Freq,
+                channel(Freq, State#state.reg_freq_list),
+                maps:get(<<"datr">>, Packet)
+            );
+        {_Type, _RoutingInfo} ->
+            %% normally packets here would be send to the router
+            %% but in light mode we can just discard non poc packets
+            noop
+    end,
+    handle_packets(Tail, Gateway, RxInstantLocal_us, State#state{last_mono_us = RxInstantLocal_us, last_tmst_us = maps:get(<<"tmst">>, Packet)}).
+
+-spec route(binary()) -> any().
+ route(Pkt) ->
+    case longfi:deserialize(Pkt) of
+        error ->
+            route_non_longfi(Pkt);
```

Comment:
This is missing the code from miner_lora to handle gateway-rs packet routing offloading